subprojects/workers/src/main/java/org/gradle/workers/internal/InProcessCompilerDaemonFactory.java

```diff
@@ -0,0 +1,144 @@
+import java.io.ObjectOutputStream;
+import java.io.Serializable;
+import java.util.concurrent.Callable;
+
+public class InProcessCompilerDaemonFactory implements WorkerDaemonFactory {
+    private static final Logger LOGGER = Logging.getLogger(InProcessCompilerDaemonFactory.class);
+
+    private final ClassLoaderFactory classLoaderFactory;
+    private final BuildOperationWorkerRegistry buildOperationWorkerRegistry;
+    private final BuildOperationExecutor buildOperationExecutor;
+
+    public InProcessCompilerDaemonFactory(ClassLoaderFactory classLoaderFactory, BuildOperationWorkerRegistry buildOperationWorkerRegistry, BuildOperationExecutor buildOperationExecutor) {
+        this.classLoaderFactory = classLoaderFactory;
+        this.buildOperationWorkerRegistry = buildOperationWorkerRegistry;
+        this.buildOperationExecutor = buildOperationExecutor;
+    }
+
+    @Override
+    public WorkerDaemon getDaemon(final Class<? extends WorkerDaemonProtocol> serverImplementationClass, File workingDir, final DaemonForkOptions forkOptions) {
+        return new WorkerDaemon() {
+            @Override
+            public <T extends WorkSpec> DefaultWorkResult execute(WorkerDaemonAction<T> action, T spec) {
+                return execute(action, spec, buildOperationWorkerRegistry.getCurrent(), buildOperationExecutor.getCurrentOperation());
+            }
+
+            @Override
+            public <T extends WorkSpec> DefaultWorkResult execute(final WorkerDaemonAction<T> action, final T spec, Operation parentWorkerOperation, BuildOperationExecutor.Operation parentBuildOperation) {
+                BuildOperationWorkerRegistry.Completion workerLease = parentWorkerOperation.operationStart();
+                BuildOperationDetails buildOperation = BuildOperationDetails.displayName(action.getDescription()).parent(parentBuildOperation).build();
+                try {
+                    return buildOperationExecutor.run(buildOperation, new Transformer<DefaultWorkResult, BuildOperationContext>() {
+                        @Override
+                        public DefaultWorkResult transform(BuildOperationContext buildOperationContext) {
+                            return executeInWorkerClassLoader(serverImplementationClass, action, spec, forkOptions.getClasspath(), forkOptions.getSharedPackages());
+                        }
+                    });
+                } finally {
+                    workerLease.operationFinish();
+                }
+            }
+        };
+    }
+
+    private <T extends WorkSpec> DefaultWorkResult executeInWorkerClassLoader(Class<? extends WorkerDaemonProtocol> serverImplementationClass, WorkerDaemonAction<T> action, T spec, Iterable<File> classpath, Iterable<String> sharedPackages) {
+        ClassLoader workerClassLoader = createWorkerClassLoader(serverImplementationClass, classpath, sharedPackages);
+        try {
+            LOGGER.info("Executing {} in in-process worker.", action.getDescription());
```

Comment:
Hmmm...not sure what the best rule of thumb should be for these logging statements.  Personally, I think this is probably too much logging for info level, but that's entirely subjective.  WDYT?

I totally agree that we should make them consistent, though.